amethyst_renderer/src/error.rs

```diff
@@ -1,126 +1,131 @@
 //! Renderer error types.
 
-use std::error::Error as StdError;
-use std::fmt::{Display, Formatter};
-use std::fmt::Result as FmtResult;
-use std::result::Result as StdResult;
-
 use gfx;
 use gfx_core;
+use glutin;
 
-/// Renderer result type.
-pub type Result<T> = StdResult<T, Error>;
-
-/// Common renderer error type.
-#[derive(Debug)]
-pub enum Error {
-    /// Failed to create a buffer.
-    BufferCreation(gfx::buffer::CreationError),
-    /// A render target with the given name does not exist.
-    NoSuchTarget(String),
-    /// Failed to initialize a render pass.
-    PassInit(gfx::PipelineStateError<String>),
-    /// Failed to create a pipeline state object (PSO).
-    PipelineCreation(gfx_core::pso::CreationError),
-    /// Failed to create thread pool.
-    PoolCreation(String),
-    /// Failed to create and link a shader program.
-    ProgramCreation(gfx::shade::ProgramError),
-    /// Failed to create a resource view.
-    ResViewCreation(gfx::ResourceViewError),
-    /// Failed to create a render target.
-    TargetCreation(gfx::CombinedError),
-    /// Failed to create a texture resource.
-    TextureCreation(gfx::texture::CreationError),
-    /// The window handle associated with the renderer has been destroyed.
-    WindowDestroyed,
-}
-
-impl StdError for Error {
-    fn description(&self) -> &str {
-        match *self {
-            Error::BufferCreation(_) => "Failed to create buffer!",
-            Error::NoSuchTarget(_) => "Target with this name does not exist!",
-            Error::PassInit(_) => "Failed to initialize render pass!",
-            Error::PipelineCreation(_) => "Failed to create PSO!",
-            Error::PoolCreation(_) => "Failed to create thread pool!",
-            Error::ProgramCreation(_) => "Failed to create shader program!",
-            Error::ResViewCreation(_) => "Failed to create resource view!",
-            Error::TargetCreation(_) => "Failed to create render target!",
-            Error::TextureCreation(_) => "Failed to create texture!",
-            Error::WindowDestroyed => "Window has been destroyed!",
+error_chain! {
+    errors {
+        /// Failed to create a buffer.
+        BufferCreation(e: gfx::buffer::CreationError) {
```

Comment:
According to docs [here](https://docs.rs/error-chain/0.11.0-rc.2/error_chain/index.html)
when you wrap another error (not a chain) you put it into `foreign_links` block.
This will add `From` implementations automatically.

Reply:
If you will wrap this chain in main crate you need to add it in `links` block

Reply:
If you use foreign_links, you gain ergonomics, but lose the ability to control the description (description is forwarded from contained error). I'll implement it using foreign_links, but want to note the trade-off.